File: poc-kronos-frontend/src/model/funcionario/FuncionarioFuncao.js
class FuncionarioFuncao {
  constructor(dados) {
    this.id = dados ? dados.id : null;
    this.idFuncionario = dados ? dados.idFuncionario : null;
    this.funcaoRegistro = dados ? dados.funcaoRegistro : null;
    this.idTipoCategoriaFuncao = dados ? dados.idTipoCategoriaFuncao : null;
    this.idTipoFuncao = dados ? dados.idTipoFuncao : null;
    this.funcao = dados ? dados.funcao : null;
    this.categoria = dados ? dados.categoria : null;
    this.ativo = dados ? dados.ativo : null;
    this.nomeTipoFuncao = dados ? dados.nomeTipoFuncao : null;
    this.funcaoSelecionada = dados
      ? dados.idFuncionario
        ? true
        : false
      : false;
  }
}
export default FuncionarioFuncao;

File: src/main/java/br/com/kronos/backend/aplicacao/funcionario/FuncionarioFuncao.java
package br.com.kronos.backend.aplicacao.funcionario;

import static br.com.kronos.backend.aplicacao.validador.Validacoes.exigirNaoNulo;

import lombok.Builder;
import lombok.Getter;

@Getter
public class FuncionarioFuncao {
	
	public static final String CACHE_NAME = "funcionarios_funcoes";
    
	private Long id;
    private Long idFuncionario;
    private int idTipoFuncao;
    private boolean funcaoRegistro;
    private boolean ativo;

	@Builder
	public FuncionarioFuncao(Long id, Long idFuncionario, int idTipoFuncao, boolean funcaoRegistro, boolean ativo) {

		super();
        this.id = id;
        this.idFuncionario = exigirNaoNulo(idFuncionario, "Funcionário");
        this.idTipoFuncao = exigirNaoNulo(idTipoFuncao, "Tipo de função");
        this.funcaoRegistro = exigirNaoNulo(funcaoRegistro, "Função de registro"); 
        this.ativo = exigirNaoNulo(ativo, "Ativo");
       
	}
}